src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3001,6 +3001,8 @@ flush_result ntp_archiver::flush() {
 }
 
 ss::future<wait_result> ntp_archiver::wait(model::offset o) {
+    ss::gate::holder holder(_gate);
```

Comment:
`flush` may take some time, are we certain that when partition is stopped the `flush` will be interrupted ?

Reply:
The only `co_await` in this function is `co_await _flush_cond.wait()`. When the gate is closed `_flush_cond` is already broken, see [there](https://github.com/redpanda-data/redpanda/pull/24780/files/f677cc34f585e0ee28b93c4a2bc9e7ab94b085d2..86f894b90ec752d926d200cfc99eacd0d11d263b#diff-1497192be88ee6c90154877e16f51c3b5d241969b5394d0b7598f6d7e30f09c0R1205-L1206) (it took GH frontend 30 seconds to load this link in my browser!)